Use JsonValue for post type supports field

oguzkocer opened the pull request:
Addresses a breaking change discovered in #1047: WordPress 6.9 returns post type `supports` as either `bool` or array, so we use `JsonValue`.

oguzkocer committed:
Use JsonValue for post type supports field

WordPress 6.9 returns post type supports as either bool or array,
so we use JsonValue wrapped in PostTypeSupportsValue for flexibility.

Changes:
- Add PostTypeSupportsValue type wrapping JsonValue
- Update PostTypeSupportsMap to use Arc<PostTypeSupportsValue>
- Update test assertion to use as_json_bool() method

oguzkocer committed:
Update Kotlin test to use asJsonBool() for post type supports

jkmassel reviewed (approved):
Thanks for taking care of this – can we validate that we have a `post_type_supports("editor")` method somewhere that just checks for that key to be present? It seems like the actual data is pretty much a black box, so we're really just checking for the key to be part of the dictionary or not.

oguzkocer committed:
Add post_type_supports helper to check feature support

Add a `supports` method to `PostTypeSupportsMap` for Rust usage and
a `post_type_supports` free function for FFI clients to check if a
post type supports a specific feature by key presence.

Includes documentation explaining the assumption that key presence
implies support, regardless of the associated value.

oguzkocer commented:
> can we validate that we have a post_type_supports("editor") method somewhere that just checks for that key to be present? It seems like the actual data is pretty much a black box, so we're really just checking for the key to be part of the dictionary or not.

@jkmassel This should be addressed in bb2ccd9005afb99c9015e13e93f049fbb0f4177c. Let me know if this works for you or not.

Note that there is some weirdness to this design because we are basically bypassing the possible `bool` value, so I've added a comment for the assumption.

oguzkocer committed:
Replace PostTypeSupportsValue with JsonValue directly

Simplify post type supports by using JsonValue directly instead of
the PostTypeSupportsValue wrapper type. Tests now use the supports
method on PostTypeSupportsMap to check feature support.

Changes:
- Change PostTypeSupportsMap.map from HashMap<_, Arc<PostTypeSupportsValue>> to HashMap<_, JsonValue>
- Remove PostTypeSupportsValue struct and its impl block
- Update Rust test to use supports method directly
- Update Kotlin test to use postTypeSupports helper

jkmassel committed:
Add Swift support
